datasets/narrativeqa_manual/narrativeqa_manual.py

```diff
@@ -0,0 +1,248 @@
+_URL = "https://github.com/deepmind/narrativeqa"
+_URLS = {
+    "documents": "https://raw.githubusercontent.com/deepmind/narrativeqa/master/documents.csv",
+    "summaries": "https://raw.githubusercontent.com/deepmind/narrativeqa/master/third_party/wikipedia/summaries.csv",
+    "qaps": "https://raw.githubusercontent.com/deepmind/narrativeqa/master/qaps.csv",
+}
+
+
+class NarrativeqaManual(datasets.GeneratorBasedBuilder):
+    """The NarrativeQA Manual dataset"""
+
+    VERSION = datasets.Version("1.0.0")
+
+    @property
+    def manual_download_instructions(self):
+        return """ You need to manually download the stories for this dataset using the script provided by the authors \
+                (https://github.com/deepmind/narrativeqa/blob/master/download_stories.sh). Running the shell script creates a folder named "tmp"\
+                in the root directory and downloads the stories there. This folder containing the stories\
+                can be used to load the dataset via `datasets.load_dataset("narrativeqa_manual", data_dir="<path/to/folder>")."""
+
+    def _info(self):
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "kind": datasets.Value("string"),
+                    "story_url": datasets.Value("string"),
+                    "story_file_size": datasets.Value("string"),
+                    "wiki_url": datasets.Value("string"),
+                    "wiki_title": datasets.Value("string"),
+                    "story_word_count": datasets.Value("string"),
+                    "story_start": datasets.Value("string"),
+                    "story_end": datasets.Value("string"),
+                    "story_text": datasets.Value("string"),
+                    "summary": datasets.Value("string"),
+                    "summary_tokenized": datasets.Value("string"),
+                    "qaps": datasets.features.Sequence(
+                        {
+                            "question": datasets.Value("string"),
+                            "answer1": datasets.Value("string"),
+                            "answer2": datasets.Value("string"),
+                            "question_tokenized": datasets.Value("string"),
+                            "answer1_tokenized": datasets.Value("string"),
+                            "answer2_tokenized": datasets.Value("string"),
+                        }
+                    ),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URLS)
+        manual_dir = os.path.abspath(os.path.expanduser(dl_manager.manual_dir))
+
```

Comment:
```suggestion
        if not os.path.exists(manual_dir):
            raise FileNotFoundError(
                "{} does not exist. Make sure you insert a manual dir via `datasets.load_dataset('narrativeqa_manual', data_dir=...)` that includes the stories downloaded from the original repository. Manual download instructions: {}".format(
                    manual_dir, self.manual_download_instructions
                )
            )
```

Could you move the check on the `manual_dir` here please ? This way it's triggered earlier before trying to compute anything

Reply:
Ok I've done this.